### DISCUSSION TITLE
Os.js with vscode?

### DISCUSSION THREAD

**Discussion by ghost** [os-js/OS.js#819] [Q&A] [answered]
Hi all.

#### 1. concept
I was thinking of developing a text/code editor similar to vscode for os.js. Recently, I saw this repository called [orta/Essence](https://github.com/orta/Essence), [DTStack/molecule](https://github.com/DTStack/molecule) or [TiaGoiNsaNy/clone-layout-vscode](https://github.com/TiaGoiNsaNy/clone-layout-vscode). Please read this [greedybrain/i-used-vscode-itself-as-my-portfolio](https://dev.to/greedybrain/i-used-vscode-itself-as-my-portfolio-1fg5) and see this [dtstack.github.io/molecule-examples](https://dtstack.github.io/molecule-examples/#/)

#### 1.1 what is "orta/Essence", "DTStack/molecule", "TiaGoiNsaNy/clone-layout-vscode" or [bhongy/vscode-react-clone](https://github.com/bhongy/vscode-react-clone)?
A VSCode UI.

#### 2. idea
![image](https://user-images.githubusercontent.com/111701513/206926948-401ac9df-1318-4cb0-94e4-d8897220f446.png)

#### 3. why?
1. My goal in this post is to see if there is a need or interest in the os.js community to have an editor similar to vscode.
2. I was thinking of basing with "orta/Essence", "DTStack/molecule" or "TiaGoiNsaNy/clone-layout-vscode" to create a similar vscode in os.js.

#### 4. Alternative?
![image](https://user-images.githubusercontent.com/111701513/207051780-6fa0f82f-eb73-49c5-bcd2-dc8fc5a136e3.png)

**image-description:** If vscode is not a good option, there is an alternative like gnome builder.

#### 5. question/feedback
- *Have you all thought about this idea?*
- Do you all find this idea interesting?

**Comment by andersevenrud** [comment, marked as answer]
I actually created an alternative to Ace using Monaco, which is the same
thing. But never uploaded it to Github.

So this is fully possible and a good idea since VSCode is a much better
alternative and would be a nice replacement.

**Reply by ghost** [reply]
Hi andersevenrud, thank you for feedback.
